lib/src/realm_object.dart

```diff
@@ -309,10 +333,24 @@ mixin RealmObject on RealmEntity implements Finalizable {
       return get(this, name);
     }
 
+    if (invocation.isSetter) {
+      final name = _symbolRegex.firstMatch(invocation.memberName.toString())?.namedGroup("symbolName");
+      if (name == null) {
+        throw RealmError(
+            "Could not find symbol name for ${invocation.memberName}. This is likely a bug in the Realm SDK - please file an issue at https://github.com/realm/realm-dart/issues");
+      }
+
+      return set(this, name, invocation.positionalArguments.single);
+    }
```

Comment:
I needed this for a test - in a follow-up PR we can add more tests and announce this as a supported functionality.